lib/note-editor.jsx

```diff
@@ -2,14 +2,28 @@ import React, { PropTypes } from 'react';
 import { connect } from 'react-redux';
 import classNames from 'classnames';
 import showdown from 'showdown';
-import xssFilter from 'showdown-xss-filter';
+import xssfilter from 'showdown-xss-config';
 import NoteDetail from './note-detail';
 import TagField from './tag-field';
 import NoteToolbar from './note-toolbar';
 import RevisionSelector from './revision-selector';
 import { get, property } from 'lodash';
 
-const markdownConverter = new showdown.Converter({ extensions: [xssFilter] });
+// whitelist <input> (only checkbox type) and <li> tags
+const xssConfig = {
+  onTag(tag, html) {
+    if (tag === 'input') {
+      if (html.includes('type="checkbox"')) return html;
```

Comment:
we would definitely want to lock this down. right now it appears like if the input is a checkbox then we'll also allow XSS attacks in other unrelated fields of the same HTML

Reply:
Can you articulate more? I'm very interested if I missed something.
Without whitelisting this, rendering checkboxes is near impossible.
Also, stricly allowing only checkbox type input seems pretty harmless..

Reply:
I can actually improve sanitization of <input>, stripping it from all the non-type attributes, if that was the problem.

Like, enforcing that only ```<input type="checkbox disabled>``` are returned.